datafusion/sql/src/expr/mod.rs

```diff
@@ -629,6 +630,36 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         }
     }
 
+    /// Parses a struct(..) expression and plans it creation
+    fn parse_struct(
+        &self,
+        schema: &DFSchema,
+        planner_context: &mut PlannerContext,
+        values: Vec<sqlparser::ast::Expr>,
+        fields: Vec<StructField>,
+    ) -> Result<Expr> {
+        if !fields.is_empty() {
+            return not_impl_err!("Struct fields are not supported yet");
+        }
+        let is_named_struct = values
+            .iter()
+            .any(|value| matches!(value, SQLExpr::Named { .. }));
+
+        let mut create_struct_args = if is_named_struct {
+            self.create_named_struct_expr(values, schema, planner_context)?
+        } else {
+            self.create_struct_expr(values, schema, planner_context)?
+        };
+
+        for planner in self.planners.iter() {
+            match planner.plan_struct_literal(create_struct_args, is_named_struct)? {
+                PlannerResult::Planned(expr) => return Ok(expr),
+                PlannerResult::Original(args) => create_struct_args = args,
+            }
+        }
```

Comment:
> Can we have one StructPlan that returns `struct` or `named_struct` based on the values?
> 
> ```rust
> let mut create_struct_args = if is_named_struct {
>             self.create_named_struct_expr(values, schema, planner_context)?
>         } else {
>             self.create_struct_expr(values, schema, planner_context)?
>         };
> ```
> 
> The logic here should be inside planner

@jayzhan211 Is this what u suggesting ? 
If you were suggesting to move `create_struct_expr` & `create_named_struct_expr` to planner. I see an issues over there i.e these functions intnerally call `self.sql_expr_to_logical_expr`. I would rather have it here to keep it simple. Sorry i understood it wrong.

Reply:
Maybe like https://github.com/apache/datafusion/pull/11318 ?

Reply:
Based on my understanding we also have a `named_struct` function which is an alias for `struct` function with input expressions **optionally named**. 

In this PR i am not moving `named_struct` function to udp. 

My understanding was to move `create_named_struct` & `create_struct` which were solely handling `struct` function.

Reply:
> > Can we have one StructPlan that returns `struct` or `named_struct` based on the values?
> > ```rust
> > let mut create_struct_args = if is_named_struct {
> >             self.create_named_struct_expr(values, schema, planner_context)?
> >         } else {
> >             self.create_struct_expr(values, schema, planner_context)?
> >         };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The logic here should be inside planner
> 
> @jayzhan211 Is this what u suggesting ?
> 
> If you were suggesting to move `create_struct_expr` & `create_named_struct_expr` to planner. I see an issues over there i.e these functions intnerally call `self.sql_expr_to_logical_expr`. I would rather have it here to keep it simple. Sorry if i understood it wrong.

Right, they are not possible to move into planner. Another thing we can improve on is, create c0, c1...cn for struct  before `plan_struct_literal` then we can call named_struct here

Reply:
Liked mentioned earlier. Thats a great idea, i will make the change in another PR. Thanks for your help @jayzhan211